2000-12-29  Jakub Jelinek  <jakub@redhat.com>

	* decl.c (init_decl_processing): Fix sign of wchar_type_node.

--- gcc/cp/decl.c.jj	Fri Dec 29 15:41:50 2000
+++ gcc/cp/decl.c	Fri Dec 29 15:46:23 2000
@@ -6545,9 +6545,9 @@ init_decl_processing ()
   wchar_type_node = TREE_TYPE (IDENTIFIER_GLOBAL_VALUE (wchar_type_node));
   wchar_type_size = TYPE_PRECISION (wchar_type_node);
   if (TREE_UNSIGNED (wchar_type_node))
-    wchar_type_node = make_signed_type (wchar_type_size);
-  else
     wchar_type_node = make_unsigned_type (wchar_type_size);
+  else
+    wchar_type_node = make_signed_type (wchar_type_size);
   record_builtin_type (RID_WCHAR, "__wchar_t", wchar_type_node);
 
   /* Artificial declaration of wchar_t -- can be bashed */
